src/Xamarin.Android.Build.Tasks/Tasks/ResolveAssemblies.cs

```diff
@@ -275,14 +283,21 @@ void CheckAssemblyAttributes (AssemblyDefinition assembly, MetadataReader reader
 					case "System.Runtime.Versioning.TargetFrameworkAttribute": {
 							var arguments = attribute.GetCustomAttributeArguments ();
 							foreach (var p in arguments.FixedArguments) {
+								// Of the form "MonoAndroid,Version=v8.1"
 								var value = p.Value?.ToString ();
-								if (value != null && value.StartsWith ("MonoAndroid", StringComparison.Ordinal)) {
-									var values = value.Split ('=');
-									var apiLevel = MonoAndroidHelper.SupportedVersions.GetApiLevelFromFrameworkVersion (values [1]);
-									if (apiLevel != null) {
-										var assemblyName = reader.GetString (assembly.Name);
-										Log.LogDebugMessage ("{0}={1}", assemblyName, apiLevel);
-										api_levels [assemblyName] = apiLevel.Value;
+								if (!string.IsNullOrEmpty (value)) {
+									var commaSeparated = value.Split (',');
+									targetFrameworkIdentifier = commaSeparated [0];
+									if (targetFrameworkIdentifier == "MonoAndroid" && commaSeparated.Length > 1) {
+										var values = commaSeparated [1].Split ('=');
```

Comment:
For that matter, I'm not sure we actually need to split on `=` either.  We "know" that the `TargetFrameworkAttribute` value ends with the version, e.g.

```csharp
[assembly: global::System.Runtime.Versioning.TargetFrameworkAttribute("MonoAndroid,Version=v9.0",
    FrameworkDisplayName = "Xamarin.Android v9.0 Support")]
```

Thus, we could do a `value.IndexOf("Version=")` and Substring after the `Version=` string.  This would avoid the `.Split()` (and associated memory use).